File: Lua/Beginner/1021.lua
valor = io.read('*n')

notas = {100, 50, 20, 10, 5, 2}
moedas = {1, 0.5, 0.25, 0.1, 0.05, 0.01}

print('NOTAS:')

for _, nota in pairs(notas) do
	quantidade = 0
	
	while valor >= nota do
		valor = valor - nota
		quantidade = quantidade + 1
	end
	
	print(string.format('%d nota(s) de R$ %.2f', quantidade, nota))
end

print('MOEDAS:')

for _, moeda in pairs(moedas) do
	quantidade = 0
	
	while valor >= moeda do
		valor = valor - moeda
		quantidade = quantidade + 1
	end
	
	print(string.format('%d moeda(s) de R$ %.2f', quantidade, moeda))
end

File: Rust/Iniciante/1021.rs
use std::io;
use std::error::Error;

fn main() -> Result<(), Box<dyn Error>> {
	let stdin = io::stdin();
	let mut buffer = String::new();

	let notas: Vec<i64> = vec![10000, 5000, 2000, 1000, 500, 200];
	let moedas: Vec<i64> = vec![100, 50, 25, 10, 5, 1];

	stdin.read_line(&mut buffer)?;
	let valor: f64 = buffer.trim().parse()?;
	let mut valor: i64 = (valor * 100.0) as i64;

	println!("NOTAS:");
	for nota in notas {
		let qt: i64 = valor / nota;
		valor %= nota;

		println!("{} nota(s) de R$ {:.2}", qt, nota as f64 / 100.0);
	}

	println!("MOEDAS:");
	for moeda in moedas {
		let qt: i64 = valor / moeda;
		valor %= moeda;

		println!("{} moeda(s) de R$ {:.2}", qt, moeda as f64 / 100.0);
	}

	Ok(())
}